src/eigen/lobpcg_hyper_impl.jl

```diff
@@ -165,6 +164,46 @@ function ortho(X, Y, BY; tol=2eps(real(eltype(X))))
 end
 
 
+# Given A as a list of blocks [A1, A2, A3] forms the matrix-matrix product
+# A' * B avoiding a concatenation of the blocks to a dense array
```

Comment:
this is A * B not A' * B right? Maybe mul and adjmul then?

Reply:
Move these two functions at the beginning of the file, and explain why they are needed (that we store the subspaces Y =[X, AX, P] as tuples instead of explicit structs)

Reply:
Also maybe [A1, A2, ...]? To show the size is variable

Reply:
Good point. I'd keep the block prefix, however, to distinguish them better from "mul!" but I would be ok with bmul for example.

Reply:
yes it does A * B

Reply:
bmul and boverlap?

Reply:
sounds good, I'll go for that.